### ISSUE TITLE
use_self false positive

### ISSUE THREAD

**Issue by bluetech** [rust-lang-nursery/rust-clippy#3425]
```
clippy 0.0.212 (d8b4269 2018-11-04)
rustc 1.32.0-nightly (36a50c29f 2018-11-09)
(2018 edition)
```

This is a regression - started to happen after a rustup.

Given this code (made to look realistic):

```rust
enum Enum {
    Field,
}
impl std::fmt::Display for Enum {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
        use self::Enum::*;
        match self {
            Field => write!(f, "Field"),
        }
    }
}
```

clippy flags `use self::Enum::*;` with `use_self` and suggests `use Self::*;` which isn't valid Rust it seems ("Use of undeclared type or module `Self`").

**Comment by flip1995** [comment]
This may be caused by #3357. This lint should probably filter items on which it should lint (not on `use` statements at least).
